x-pack/plugins/security_solution/public/detections/components/rules/rule_switch/index.tsx

```diff
@@ -56,6 +57,7 @@ export const RuleSwitchComponent = ({
   const [myEnabled, setMyEnabled] = useState(enabled ?? false);
   const [, dispatchToaster] = useStateToaster();
   const rulesTableContext = useRulesTableContextOptional();
+  const invalidateRules = useInvalidateRules();
```

Comment:
If we keep the invalidation logic, I'd suggest exposing `invalidateRules` from `rulesTableContext` to make the usage easier.

Reply:
No, `invalidateRules` should work in places where the rules table context is not available, as the rule editing page.